docs/design/sql/20-match-jet-job-to-sql.md

```diff
@@ -0,0 +1,92 @@
+# Matching the SQL query to the Jet job
+
+|||
+|---|---|
+|Related Jira|[HZ-2859](https://hazelcast.atlassian.net/browse/HZ-2859)|
+|Document Status / Completeness|IN PROGRESS|
+|Requirement owner|Saumitra Sahi|
+|Developer(s)|Sasha Syrotenko, Ivan Yaschishin|
+|Quality Engineer|Isaac Sumner|
+|Technical Reviewers|Krzystof Jamroz|
+|Version|5.4|
+
+## Background
+In Hazelcast Platform 5.3, the sole SQL job metrics report can be accessed exclusively through
+the `getJobAndSqlSummaryList()` method within the Java client. This approach doesn't align with
+the idea of SQL serving as the primary Hazelcast interface. Consequently, there is currently
+no ability to monitor the status of a running interactive SQL query, because it's impossible
+to associate it with the corresponding Jet job.
+
+To address this issue, we aim to enable the match of interactive SQL queries with their running Jet jobs,
+allowing for the collection of a wider range of available metrics.
```

Comment:
this TDD does not add any new metrics. It will make easier to "show metrics/live metrics for a query I just started"